SConstruct

```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 
-EnsureSConsVersion(3, 0, 0)
+EnsureSConsVersion(4, 2, 0)
```

Comment:
That's a bit too steep, we still support building Godot on fairly old distros which may not have SCons 4.2.0 packaged yet (as a matter of fact almost no distro has it packaged *right now* - only Arch Linux and Fedora).

There are Godot packages in Linux distro and they rely on system SCons, so they can't just pull it from the pip repos.

Instead, you should check the SCons version at runtime to conditionally support ninja build for SCons 4.2.0 or later, as we do for the compilation DB:

https://github.com/godotengine/godot/blob/3ca2a48767486a52bf4ec5b784a25a1beb2a3f50/SConstruct#L355-L362

(Might also be worth considering if this API or a similar one could be promoted to public in SCons upstream.)

Reply:
Okay that's reasonable. I'll add a similar conditional check and file something for something more standard upstream.

Reply:
add check like so in
e3b3d25